src/uncss.js

```diff
@@ -206,23 +205,30 @@ function init(files, options, callback) {
         raw: null
     });
 
-    process(options, callback);
+    process(options).then(([css, report]) => callback(null, css, report), callback);
 }
 
 function processAsPostCss(files, options, pages) {
     return uncss(pages, options.rawPostCss, options.ignore);
 }
 
-function process(opts, callback) {
+function process(opts) {
     const resource = getHTML(opts.html, opts);
-    return Promise.using(resource, (pages) => {
+    return resource.then((pages) => {
         if (opts.usePostCssInternal) {
-            return processAsPostCss(opts.files, opts, pages);
+            return processAsPostCss(opts.files, opts, pages)
+            .then(cleanup);
         }
         return getStylesheets(opts.files, opts, pages)
-          .spread(getCSS)
-          .spread(processWithTextApi);
-    }).asCallback(callback, { spread: !opts.usePostCssInternal });
+          .then(getCSS)
+          .then(processWithTextApi)
+          .then(cleanup);
+
+        function cleanup (result) {
```

Comment:
Shouldn't this cleanup callback be on line 218, before the attempts to reference it? Or does it work properly down below, too?

Reply:
`function XXX` declarations are hoisted, along with their values, so this works great. You can't do the same thing if you're doing `const xxx = function () { ... }`.

Reply:
Ahh interesting. I have a small preference for putting the cleanup function earlier, but other than that, these changes look good to me! (Sorry, but no Windows boxes available, so can't help with that... :/ )